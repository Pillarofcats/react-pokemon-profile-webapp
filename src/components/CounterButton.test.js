import React from 'react';
import {shallow} from 'enzyme';
import CounterButton from "./CounterButton.js";

it("Expect to Render the Counter Button", () => {
  const mockButtonColor = 'red';
  expect(shallow(<CounterButton color={mockButtonColor}/>)).toMatchSnapshot();
});

it("Correctly increments the Counter", () => {
  const mockButtonColor = 'red';
  const wrapper = shallow(<CounterButton color={mockButtonColor}/>);

  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2});
  expect(wrapper.props().color).toEqual('red');
});