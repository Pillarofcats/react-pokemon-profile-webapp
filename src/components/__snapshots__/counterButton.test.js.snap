// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to Render the Counter Button 1`] = `ShallowWrapper {}`;
